core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -445,13 +446,13 @@ private ClusterCacheStatus initCacheStatusIfAbsent(String cacheName, CacheMode c
             };
          }
          AvailabilityStrategy availabilityStrategy;
-         Configuration cacheConfiguration = cacheManager.getCacheConfiguration(cacheName);
-         PartitionHandling partitionHandling = cacheConfiguration != null ? cacheConfiguration.clustering().partitionHandling().whenSplit() : null;
-         boolean resolveConflictsOnMerge = partitionHandling == PartitionHandling.ALLOW_READ_WRITES && cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
+         Configuration config = cacheManager.getCacheConfiguration(cacheName);
```

Comment:
I'm not sure how this code's semantics are different to what we previously had? Or is there an issue with how it was previously?

Reply:
Yeah, it's an existing issue. Actually there was a comment just before the `getCacheConfiguration` line that we assume the configuration to be defined everywhere before the cache is first started anywhere, but Radim's stuff slipped in between and the comment wasn't visible in the diff :)

TBH I'm not sure why I haven't added the partition handling stuff to `CacheJoinInfo`. Maybe it's because there was already other stuff in the partition handling code that needed the cache to be defined everywhere, or maybe we just decided that `partitionHandling(true)` is a new feature and doesn't have to support asymmetric caches. Either way, now I think it would have been better to add it to `CacheJoinInfo` anyway and throw an exception if the cache is not defined on the coordinator.

Reply:
Ok, I have created [ISPN-8613](https://issues.jboss.org/browse/ISPN-8613). Lets tackle this one after Beta2